### ISSUE TITLE
pyright should reject `*Ts` as a type param in a `TypeAliasType` definition

### ISSUE THREAD

**Issue by hamdanal** [microsoft/pyright#7284]
It is not really called out in PEP 695 but it seems it should be rejected. Consider this example:

```python
from typing_extensions import TypeAliasType, TypeVarTuple, Unpack

Ts = TypeVarTuple("Ts")

Alias1 = TypeAliasType("Alias1", tuple[*Ts], type_params=(Ts,))  # pyright: OK
Alias2 = TypeAliasType("Alias2", tuple[*Ts], type_params=(*Ts,))  # pyright: OK <- false negative?
Alias3 = TypeAliasType("Alias3", tuple[*Ts], type_params=(Unpack[Ts],))  # pyright: Error
```

I noticed this while adding support to `TypeAliasType` in mypy https://github.com/python/mypy/pull/16926 which in its current state rejects `Alias2` and `Alias3` above while pyright only emits an error on `Alias3` usage.

**versions:**
pyright 1.1.350
python 3.11

**Comment by erictraut** [comment]
Thanks for filing the issue. I agree this is a false negative condition.

It will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.351, which I just published.
